### ISSUE TITLE
Some rules for prop-types are false positive in switch/case component

### ISSUE THREAD

**Issue by g4g1** [yannickcr/eslint-plugin-react#2118]
Example:
```
import PropTypes from 'prop-types';
import React from 'react';

const MyComponent= (props) => {
  switch (props.usedProp) {
    case 1:
      return (<div />);
    default:
      return <div />;
  }
};

MyComponent.propTypes = {
  unUsedProp: PropTypes.string,
};

MyComponent.defaultProps = {
};

export default MyComponent;
```
Code should throw 3 errors:
react/prop-types (usedProp missing in props)
react/required-default-props (for unUsedProp)
react/no-unused-prop-types (for unUsedProp)
but it seems ok for eslint
